pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2161,6 +2161,62 @@ func (a *ApplicationBackupController) cleanupResources(
 		log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
 		return err
 	}
+	// Check if the backup is of the type virtual machine.
+	if IsBackupObjectTypeVirtualMachine(backup) {
+		preExecRule := backup.Spec.PreExecRule
+		postExecRule := backup.Spec.PostExecRule
+	
+		deleteRuleIfExists := func(ruleName string) error {
+			// Fetch the rule first
+			rule, err := storkops.Instance().GetRule(ruleName, backup.Namespace)
+			if err != nil {
+				if k8s_errors.IsNotFound(err) {
+					// Log and skip deletion for not found rule
+					logrus.Infof("rule CR [%v] not found, skipping deletion: %v", ruleName, err)
+					return nil
+				}
+				// Handle other errors
+				errMsg := fmt.Sprintf("failed to retrieve the rule CR [%v]: %v", ruleName, err)
+				log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
+				return err
+			}
+			// Check if annotations match and rule is not nil
+			if rule != nil &&
+				rule.Annotations[backupUIDKey] == string(backup.UID) &&
+				rule.Annotations[createdByKey] == createdByValue {
+				// Log before deletion
+				logrus.Infof("deleting rule CR: %v", rule.Name)
+				// Delete the rule
+				err := storkops.Instance().DeleteRule(rule.Name, backup.Namespace)
+				if err != nil && !k8s_errors.IsNotFound(err) {
+					// Log after deletion, include error details
+					errMsg := fmt.Sprintf("failed to delete the rule CR [%v]: %v", rule.Name, err)
+					log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
+					return err
+				}
+				logrus.Infof("rule CR deleted successfully: %v", rule.Name)
+			}
+	
+			return nil
+		}
+	
+		// Delete the pre-exec rule if not empty or null
+		if preExecRule != "" {
+			logrus.Infof("deleting the preExec rule: %v", preExecRule)
+			if err := deleteRuleIfExists(preExecRule); err != nil {
+				return err
+			}
+		}
+	
+		// Delete the post-exec rule if not empty or null
```

Comment:
Can you please remove comments that are not required?

Reply:
Removed the unnecessary comments